apps/managedidentity/managedidentity_test.go

```diff
@@ -4,43 +4,401 @@ package managedidentity
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
 	"testing"
+	"time"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/errors"
+	internalTime "github.com/AzureAD/microsoft-authentication-library-for-go/apps/internal/json/types/time"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/internal/oauth/ops/accesstokens"
+)
+
+const (
+	// test Resources
+	resource              = "https://demo.azure.com"
+	resourceDefaultSuffix = "https://demo.azure.com/.default"
 )
 
-func fakeClient(mangedIdentityId ID, options ...Option) (Client, error) {
-	client, err := New(mangedIdentityId, options...)
+type HttpRequest struct {
+	Source   Source
+	Resource string
+	Identity ID
+}
+
+type SuccessfulResponse struct {
+	AccessToken string `json:"access_token"`
+	ExpiresOn   int64  `json:"expires_on"`
+	Resource    string `json:"resource"`
+	TokenType   string `json:"token_type"`
+	ClientID    string `json:"client_id"`
+}
+
+type ErrorResponse struct {
+	StatusCode    int    `json:"statusCode"`
+	Message       string `json:"message"`
+	CorrelationID string `json:"correlationId,omitempty"`
+}
+
+type fakeClient struct{}
+type errorClient struct {
+	errResponse ErrorResponse
+}
+
+func fakeMIClient(mangedIdentityId ID, options ...ClientOption) (Client, error) {
+	fakeClient, err := New(mangedIdentityId, options...)
 
 	if err != nil {
 		return Client{}, err
 	}
 
-	return client, nil
+	return fakeClient, nil
 }
 
-func TestManagedIdentity(t *testing.T) {
-	client, err := fakeClient(SystemAssigned())
+func (*fakeClient) CloseIdleConnections()  {}
+func (*errorClient) CloseIdleConnections() {}
 
-	if err != nil {
-		t.Fatal(err)
+func (*fakeClient) Do(req *http.Request) (*http.Response, error) {
+	w := http.Response{
+		StatusCode: http.StatusOK,
+		Body:       io.NopCloser(strings.NewReader(getSuccessfulResponse(resource))),
+		Header:     make(http.Header),
+	}
+	return &w, nil
+}
+
+func (e *errorClient) Do(req *http.Request) (*http.Response, error) {
+	w := http.Response{
+		StatusCode: e.errResponse.StatusCode,
+		Body:       io.NopCloser(strings.NewReader(e.errResponse.Message)),
+		Header:     make(http.Header),
+	}
+	return &w, nil
+}
+
+func getSuccessfulResponse(resource string) string {
+	expiresOn := time.Now().Add(1 * time.Hour).Unix()
+	response := SuccessfulResponse{
+		AccessToken: "fakeToken",
+		ExpiresOn:   expiresOn,
+		Resource:    resource,
+		TokenType:   "Bearer",
+		ClientID:    "client_id",
+	}
+	jsonResponse, _ := json.Marshal(response)
```

Comment:
A test calling this function should fail when it gets a non-nil error. Test bugs may be unlikely here but not checking the error makes the eventual failure harder to understand or allows a test to pass when it should fail